packages/test/test-gc-sweep-tests/src/test/gcSweepTest.spec.ts

```diff
@@ -315,37 +276,40 @@ describeNoCompat("GC Random tests", (getTestObjectProvider) => {
         const container = getRandomContainer();
         const dataStore = await getRandomDataStoreFromContainer(container);
         const parentDataStore = await getRandomParentForDataStore(dataStore, container);
+        if (parentDataStore === undefined) {
+            return;
+        }
         unreferenceDataStore(dataStore, parentDataStore);
     };
 
-    const actionsList: (() => Promise<unknown>)[] = [
+    const actionsList: (() => Promise<void>)[] = [
         loadNewContainer,
         closeRandomContainer,
         createDataStoreForRandomContainer,
         referenceRandomDataStore,
         unreferenceRandomDataStore,
     ];
-    let testTime = 0;
-    const testEnd = 60 * 1000; // 1 minute
-    const testActions: { sleep: number; action: () => Promise<unknown>; }[] = [];
-    while (testTime < testEnd) {
-        const sleepTime: number = randomInteger(0, defaultSessionExpiryDurationMs + 1000);
-        const action = makeRandom().pick(actionsList);
-        testActions.push({ sleep: sleepTime, action });
-        testTime += sleepTime;
-    }
+    const testTime = 60 * 1000; // 1 minute
 
-    it("Create and reference and unreference datastores with multiple containers", async () => {
+    it(`Create and reference and unreference datastores with multiple containers. Seed: ${seed}`, async () => {
         overrideLogger.ignoreExpectedEventTypes({
             eventName: "fluid:telemetry:Container:ContainerClose",
             errorType: ContainerErrorType.clientSessionExpiredError,
         });
 
-        for (const action of testActions) {
-            void action.action();
-            await sleep(action.sleep);
+        const testEnd = Date.now() + testTime;
+        const errorList: any[] = [];
+        while (Date.now() < testEnd) {
+            const sleepTime: number = randomInteger(0, defaultSessionExpiryDurationMs + 1000);
+            const action = random.pick(actionsList);
+
+            action().catch((error) => {
+                errorList.push(error);
+            });
+            await sleep(sleepTime);
         }
-    });
+        assert(errorList.length === 0, `${errorList}`);
```

Comment:
Might give nicer output to do

```suggestion
        assert.deepEquals(errorList, [], "Expected no errors");
```

I guess it's actually the same info just presented differently.

Reply:
I tried this out, I'd like the errors to be printed if the test fails.